src/apps/filters/h3.c

```diff
@@ -1189,6 +1189,437 @@ SUBCOMMAND(uncompactCells,
+            curDepth--;
+            if (curDepth < 0) {
+                break;
+            }
+            strPos++;
+            // We may need to set up a new geoloop at this point. If the
+            // curDepth <= maxDepth - 2 then we increment the curLoop and get a
+            // new one set up.
+            if (curDepth <= maxDepth - 2) {
+                if (curLoop == 0) {
+                    out->geoloop.verts = verts;
+                    out->geoloop.numVerts = curVert;
+                } else {
+                    GeoLoop *holes = calloc(out->numHoles + 1, sizeof(GeoLoop));
+                    for (int i = 0; i < out->numHoles; i++) {
+                        holes[i].numVerts = out->holes[i].numVerts;
+                        holes[i].verts = out->holes[i].verts;
+                    }
+                    free(out->holes);
+                    holes[out->numHoles].verts = verts;
+                    holes[out->numHoles].numVerts = curVert;
+                    out->holes = holes;
+                    out->numHoles++;
+                }
+                curLoop++;
+                curVert = 0;
+                numVerts = 6;
+                verts = calloc(numVerts, sizeof(LatLng));
+            }
+            continue;
+        }
+        // Try to grab a floating point value followed by optional whitespace, a
+        // comma, and more optional whitespace, and a second floating point
+        // value, then store the lat, lng pair
+        double lat, lng;
+        int count = 0;
+        // Must grab the closing ] or we might accidentally parse a partial
+        // float across buffer boundaries
+        char closeBracket[2] = "]";
+        int res = sscanf(polygonString + strPos, "%lf%*[, \n]%lf%1[]]%n", &lat,
+                         &lng, &closeBracket[0], &count);
+        if (count > 0 && res == 3) {
+            verts[curVert].lat = H3_EXPORT(degsToRads)(lat);
+            verts[curVert].lng = H3_EXPORT(degsToRads)(lng);
+            curVert++;
+            // Create a new vert buffer, copy the old buffer, and free it, if
+            // necessary
+            if (curVert == numVerts) {
+                LatLng *newVerts = calloc(numVerts * 2, sizeof(LatLng));
+                for (int i = 0; i < numVerts; i++) {
+                    newVerts[i].lat = verts[i].lat;
+                    newVerts[i].lng = verts[i].lng;
+                }
+                free(verts);
+                verts = newVerts;
+                numVerts *= 2;
+            }
+            strPos += count;
+            curDepth--;
+            continue;
+        }
+        // Check for whitespace and skip it if we reach this point.
+        if (polygonString[strPos] == ',' || polygonString[strPos] == ' ' ||
+            polygonString[strPos] == '\n' || polygonString[strPos] == '\t' ||
+            polygonString[strPos] == '\r') {
+            strPos++;
+            continue;
+        } else {
+            if (strPos != 0 && fp != 0) {
```

Comment:
When you specify the string being parsed as a command line argument rather than passing the filename to read from. (This let me DRY up the parsing logic instead of having one for CLI arg and another for files.)

Reply:
Got it, thanks!